pkg/api/presenters/resource.go

```diff
@@ -61,7 +61,14 @@ func PresentResource(resource *api.Resource) (*openapi.Resource, error) {
 		DeleteOption:   deleteOption,
 		UpdateStrategy: updateStrategy,
 		Status:         status,
-	}, nil
+	}
+
+	// set the deletedAt field if the resource has been marked as deleted
+	if !resource.DeletedAt.Time.IsZero() {
+		res.DeletedAt = openapi.PtrTime(resource.DeletedAt.Time)
```

Comment:
why need this check? can we just use `DeletedAt: = openapi.PtrTime(resource.DeletedAt)`

Reply:
this should never happen, just for double check.
because 'onDelete' will only called after the `MarkAsDeleting`.